File: LeetCodeProblemsGo/LeetCode3487MaximumUniqueSubarraySumAfterDeletion/LeetCode3487MaximumUniqueSubarraySumAfterDeletion.go
package LeetCode3487MaximumUniqueSubarraySumAfterDeletion

func maxSum(nums []int) int {
	maxNum := -101
	sum := 0
	numsMap := make(map[int]any)
	for _, num := range nums {
		maxNum = max(num, maxNum)
		if num <= 0 {
			continue
		}
		if _, exists := numsMap[num]; exists {
			continue
		}
		numsMap[num] = struct{}{}
		sum += num
	}
	if maxNum <= 0 {
		return maxNum
	}
	return sum
}

File: LeetCodeProblemsCMake/Problems/LeetCode3487MaximumUniqueSubarraySumAfterDeletion.cpp
//
// Created by saeed on 7/25/25.
//

#include "LeetCode3487MaximumUniqueSubarraySumAfterDeletion.h"

#include <unordered_map>
#include <unordered_set>
using namespace std;
int LeetCode3487MaximumUniqueSubarraySumAfterDeletion::maxSum(vector<int> &nums) {
    unordered_set< int> positiveNums{};
    int sum{};
    int maxNums{-101};
    for(int num:nums) {
        if (num > maxNums) {
            maxNums = num;
        }
        if (num<=0)
            continue;
        if (positiveNums.contains(num)) {
            continue;
        }
        positiveNums.insert(num);
        sum+=num;
    }
    if (maxNums <= 0)
        return maxNums;
    return sum;
}
